### PR TITLE
Implemented pseudolabel, uda semisupervised trainers

### PR DESCRIPTION
**Pull request by arnavmdas**

### CODE REVIEW

**Comment by arnavmdas** [comment]
merge

File: LabelBench/trainer/trainer_impl/uda.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            max_probs, _ = torch.max(probs_uw, dim=-1)
+            select = max_probs.ge(self.trainer_config['p_cutoff'])
+
+            # Generate sharpened pseudolabels 
+            logits_uw = logits_uw[select]
+            pseudo_label = F.softmax(logits_uw.data.detach().cpu().float()/self.trainer_config["temperature"], dim=-1)
+
+
+        # Compute unsupervised loss
+        # log_probs_us = F.log_softmax(logits_us[select], dim=-1)
```
Comment (jifanz) [inline LabelBench/trainer/trainer_impl/uda.py:57]: Remove this line.

File: LabelBench/trainer/trainer_impl/uda.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        super().__init__(trainer_config, dataset, model_fn, model_config, metric, get_feature_fn)
+        self.use_strong = True
+        self.unlabeled_idxs = None
+        self.consistency_loss = SoftTargetCrossEntropy()
+
+    def initialize_trainer(self):
+        self.unlabeled_idxs = torch.from_numpy(self.dataset.unlabeled_idxs()).long()
+
+    def train_step(self, model, img_l, target_l, class_weights, loss_fn, idx_u, img_uw, img_us, iter):
+        # TODO: documentation
```
Comment (jifanz) [inline LabelBench/trainer/trainer_impl/uda.py:23]: I'm not sure if we need the documentation here. Perhaps you could just take the documentation from the microsoft repo or remove the TODO?

**Review by jifanz — APPROVED** [review]
LGTM!

**Pull request merged**